File: simple_listener/src/simple_listener_string.cpp
#include "simple_listener_string.hpp"


SimpleListenerString::SimpleListenerString() : Node("simple_listener_string")
{
    string_sub_ = this->create_subscription<std_msgs::msg::String>
    ("/string_msg", 10, std::bind(&SimpleListenerString::string_callback, this, std::placeholders::_1));
}

void SimpleListenerString::string_callback(const std_msgs::msg::String::SharedPtr msg)
{
    RCLCPP_INFO(rclcpp::get_logger("simple_listener_string"),
    "Subscribe : %s", msg->data.c_str());
}

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleListenerString>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: simple_listener/simple_listener/simple_listener_string.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleListenerString(Node):
    def __init__(self):
        super().__init__("simple_listener_string")
        self.string_sub_ = self.create_subscription(String, "/string_msg", self.listener_callback, 10)
        self.subscriptions

    def listener_callback(self, msg):
        self.get_logger().info("Subscribe: '%s'" % msg.data)


def main():
    rclpy.init()
    simple_listener_string = SimpleListenerString()
    rclpy.spin(simple_listener_string)
    simple_listener_string.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()
